internal/factsengine/gatherers/dir_scan_test.go

```diff
@@ -0,0 +1,207 @@
+		_, _ = tFs.Create(filePath)
+	}
+
+	_, _ = tFs.Create(fmt.Sprintf("%s/1/ASCS3", dirScanTestBasePath))
+	_, _ = tFs.Create(fmt.Sprintf("%s/1/ASDX2", dirScanTestBasePath))
+	_, _ = tFs.Create(fmt.Sprintf("%s/2/ASDX1", dirScanTestBasePath))
+
+	s.testFS = tFs
+
+}
+func (s *DirScanGathererSuite) TearDownSuite() {
+	err := s.testFS.RemoveAll(dirScanTestBasePath)
+	s.NoError(err)
+}
+
+func (s *DirScanGathererSuite) TestDirScanningErrorDirScaningWithoutGlob() {
+	groupSearcher := mocks.NewGroupSearcher(s.T())
+	groupSearcher.On("GetGroupByID", mock.AnythingOfType("string")).Return("trento", nil)
+
+	userSearcher := mocks.NewUserSearcher(s.T())
+	userSearcher.On("GetUsernameByID", mock.AnythingOfType("string")).Return("trento", nil)
+
+	g := gatherers.NewDirScanGatherer(s.testFS, userSearcher, groupSearcher)
+
+	fr := []entities.FactRequest{{
+		Argument: fmt.Sprintf("%s/1/ASCS3", dirScanTestBasePath),
+		CheckID:  "check1",
+		Gatherer: "dir_scan",
+		Name:     "dir_scan",
+	}}
+
+	result, _ := g.Gather(fr)
+	expectedResult := []entities.Fact{{
+		Name:    "dir_scan",
+		CheckID: "check1",
+		Value: &entities.FactValueMap{
+			Value: map[string]entities.FactValue{
+				"/var/test/1": &entities.FactValueMap{
+					Value: map[string]entities.FactValue{
+						"owner": &entities.FactValueString{Value: "trento"},
+						"group": &entities.FactValueString{Value: "trento"},
+						"files": &entities.FactValueList{
+							Value: []entities.FactValue{
+								&entities.FactValueMap{
+									Value: map[string]entities.FactValue{
+										"owner": &entities.FactValueString{Value: "trento"},
+										"group": &entities.FactValueString{Value: "trento"},
+										"name":  &entities.FactValueString{Value: "/var/test/1/ASCS3"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}}
+
+	s.EqualValues(expectedResult, result)
+
+}
+
+func (s *DirScanGathererSuite) TestDirScanningErrorNoArgument() {
+	g := gatherers.NewDirScanGatherer(s.testFS, &gatherers.CredentialsFetcher{}, &gatherers.CredentialsFetcher{})
+
+	fr := []entities.FactRequest{{
+		CheckID:  "check1",
+		Gatherer: "dir_scan",
+		Name:     "dir_scan",
+	}}
+
+	expectedResult := []entities.Fact{{
+		Name:    "dir_scan",
+		CheckID: "check1",
+		Value:   nil,
+		Error: &entities.FactGatheringError{
+			Type:    "dir-scan-missing-argument",
+			Message: "missing required argument",
+		},
+	}}
+
+	result, _ := g.Gather(fr)
+	s.EqualValues(expectedResult, result)
+}
+
+func (s *DirScanGathererSuite) TestDirScanningSuccess() {
+	groupSearcher := mocks.NewGroupSearcher(s.T())
+	groupSearcher.On("GetGroupByID", mock.AnythingOfType("string")).Return("trento", nil)
+
+	userSearcher := mocks.NewUserSearcher(s.T())
+	userSearcher.On("GetUsernameByID", mock.AnythingOfType("string")).Return("trento", nil)
+
+	g := gatherers.NewDirScanGatherer(s.testFS, userSearcher, groupSearcher)
+	fr := []entities.FactRequest{{
+		Argument: dirScanTestGlobPattern,
```

Comment:
Is it not better to put this `dirScanTestGlobPattern` here locally in the test?
It is not used anywhere else, and it makes the reading of the test more difficult, as you need to scroll up and down

Reply:
done